azurerm/resource_arm_hd_insight_clusters.go

```diff
@@ -0,0 +1,918 @@
+package azurerm
+
+// Code based on the terraform.provider plugin by Microsoft (R) AutoRest Code Generator.
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/services/hdinsight/mgmt/2015-03-01-preview/hdinsight"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmHDInsightClusters() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmHDInsightClustersCreate,
+		Read:   resourceArmHDInsightClustersRead,
+		Update: resourceArmHDInsightClustersUpdate,
+		Delete: resourceArmHDInsightClustersDelete,
+		Schema: map[string]*schema.Schema{
+			"cluster_type": {
+				Required: true,
+				ForceNew: true,
+				Type:     schema.TypeString,
+				ValidateFunc: validation.StringInSlice([]string{
+					"hadoop",
+					"hbase",
+					"storm",
+					"spark",
+				}, true),
+				DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+			},
+			"cluster_version": {
+				Required: true,
+				ForceNew: true,
+				Type:     schema.TypeString,
+			},
```

Comment:
I think it'd make sense to combine these into a `cluster` block:

```
cluster {
  type = "hadoop"
  version = "x.y.z"
}
```

Reply:
Yes in general we group related fields; such that it provides a better user-experience; which is why we don't 1:1 map the SDK

Reply:
I still don't think it makes sense to group `cluster` related attributes into the `cluster` sub-field which duplicates itself and will cause confusion to the customer.

Reply:
If it doesn't make sense to have the group, it doesn't make sense to have the prefix, so either it should be grouped due to the prefixing creating an implicit group or it should have the prefixes stripped if that doesn't make it lose its meaning.